libvast/vast/expression.hpp

```diff
@@ -319,6 +295,24 @@ struct predicate_transformer {
   }
 
   F f;
+
+private:
+  template <class T>
+  result_type make_result(const T& input) const {
```

Comment:
This doesn't even have to be a to be a template, as it essentially just maps the funtion `f` down the tree, so it does nothing that is specific to `conjunction` or `disjunction`. Since both of them are just simple wrappers over `std::vector<expression>`, we can simply return that here and wrap it into the right variant type at the call site. Variables of this type are called `connective` in a few places.

Reply:
Do you mind if I explore that proposed refactoring in a follow-up PR? I'll need to look at the callers a bit more to understand what's going on. As it stands, I think this PR is an improvement in the area of code, even if it's a function template.

Reply:
I realize that this is actually not that trivial because of the support for transformation that can fail.
I accept this patch as is then.